services/ingest/app/routers/auth.py

```
@@ -0,0 +1,332 @@
+@router.get("/github/callback")
+async def github_callback(
+    request: Request,
+    code: str,
+    state: str,
+    session: AsyncSession = Depends(get_session),
+) -> RedirectResponse:
+    nonce_cookie = request.cookies.get(NONCE_COOKIE)
+    try:
+        claims = jwt.decode(state, JWT_SECRET, algorithms=["HS256"])
+    except jwt.PyJWTError:
+        raise HTTPException(status_code=401, detail="Invalid or expired OAuth state")
```

Ruff (0.16.3)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)